pkg/divert/istio/divert.go

```diff
@@ -0,0 +1,222 @@
+		developerVirtualServices: map[string]*istioV1beta1.VirtualService{},
+	}
+	vsList, err := virtualservices.List(ctx, m.Deploy.Divert.Namespace, c)
+	if err != nil {
+		return nil, err
+	}
+	for i := range vsList {
+		cache.divertVirtualServices[vsList[i].Name] = vsList[i]
+	}
+
+	vsList, err = virtualservices.List(ctx, m.Namespace, c)
+	if err != nil {
+		return nil, err
+	}
+	for i := range vsList {
+		cache.developerVirtualServices[vsList[i].Name] = vsList[i]
+	}
+	return cache, nil
+}
+
+func (d *Driver) retryTranslateDeveloperVirtualService(ctx context.Context, name string) error {
+	retries := 0
+	var err error
+	for retries < 10 {
+		if _, ok := d.cache.developerVirtualServices[name]; !ok {
+			vs := createIntoDeveloperVirtualService(d.Manifest, d.cache.divertVirtualServices[name])
+			err = virtualservices.Create(ctx, vs, d.IstioClient)
+			if err == nil {
+				d.cache.developerVirtualServices[name] = vs
+				return nil
+			}
+			if !k8sErrors.IsAlreadyExists(err) {
+				return err
+			}
+			return nil
+		}
+
+		vs := translateDeveloperVirtualService(d.Manifest, d.cache.developerVirtualServices[name])
+		if vs.Annotations[model.OktetoAutoCreateAnnotation] != "" {
+			vs = createIntoDeveloperVirtualService(d.Manifest, d.cache.divertVirtualServices[name])
+			vs.ResourceVersion = d.cache.developerVirtualServices[name].ResourceVersion
+		}
+		if isEqualVirtualService(vs, d.cache.developerVirtualServices[name]) {
+			return nil
+		}
+
+		err = virtualservices.Update(ctx, vs, d.IstioClient)
+		if err == nil {
+			d.cache.developerVirtualServices[name] = vs
+			return nil
+		}
+		if !k8sErrors.IsConflict(err) {
+			return err
+		}
+		retries++
+	}
+	return err
+}
+
+func (d *Driver) retryTranslateDivertVirtualService(ctx context.Context, name string) error {
+	retries := 0
+	var err error
+	for retries < 10 {
+		vs := translateDivertVirtualService(d.Manifest, d.cache.divertVirtualServices[name])
+		if isEqualVirtualService(vs, d.cache.divertVirtualServices[name]) {
+			return nil
+		}
+
+		err = virtualservices.Update(ctx, vs, d.IstioClient)
+		if err == nil {
+			d.cache.divertVirtualServices[name] = vs
+			return nil
+		}
+		if !k8sErrors.IsConflict(err) {
+			return err
+		}
+		retries++
+	}
+	return err
+}
+
+func (d *Driver) retryRestoreDivertVirtualService(ctx context.Context, name string) error {
+	retries := 0
+	var err error
+	for retries < 10 {
+		vs := restoreDivertVirtualService(d.Manifest, d.cache.divertVirtualServices[name])
+		if isEqualVirtualService(vs, d.cache.divertVirtualServices[name]) {
+			return nil
+		}
+
+		err = virtualservices.Update(ctx, vs, d.IstioClient)
+		if err == nil {
+			d.cache.divertVirtualServices[name] = vs
+			return nil
+		}
+		if !k8sErrors.IsConflict(err) {
+			return err
+		}
+		retries++
+	}
+	return err
+}
+
+func isInDivertVirtualServices(name string, virtualServices []string) bool {
+	for _, vs := range virtualServices {
+		if vs == "*" {
```

Comment:
NIT: I think it's easier to read it with a switch case 
```suggestion
switch vs {
case "*", name:
    return true
 }
```